server/webhook.go

```diff
@@ -206,22 +216,47 @@ func (p *Plugin) permissionToProject(ctx context.Context, userID, namespace, pro
 	return true
 }
 
-func CreateHook(ctx context.Context, gitlabClient gitlab.Gitlab, info *gitlab.UserInfo, group, project string, hookOptions *gitlab.AddWebhookOptions) (*gitlab.WebhookInfo, error) {
+func (p *Plugin) createHook(ctx context.Context, gitlabClient gitlab.Gitlab, info *gitlab.UserInfo, group, project string, hookOptions *gitlab.AddWebhookOptions) (*gitlab.WebhookInfo, error) {
 	// If project scope
 	if project != "" {
-		project, err := gitlabClient.GetProject(ctx, info, group, project)
-		if err != nil {
-			return nil, err
+		var gitProject *gitlabLib.Project
+		getProjectErr := p.useGitlabClient(info, func(info *gitlab.UserInfo, token *oauth2.Token) error {
+			resp, err := p.GitlabClient.GetProject(ctx, info, token, group, project)
+			if err != nil {
+				return err
+			}
+			gitProject = resp
+			return nil
+		})
+		if getProjectErr != nil {
+			return nil, getProjectErr
 		}
-		newWebhook, err := gitlabClient.NewProjectHook(ctx, info, project.ID, hookOptions)
-		if err != nil {
-			return nil, err
+		var newWebhook *gitlab.WebhookInfo
+		getGroupErr := p.useGitlabClient(info, func(info *gitlab.UserInfo, token *oauth2.Token) error {
```

Comment:
Should we call `useGitlabClient` only once here for both operations, to avoid re-fetching the token?

Reply:
I suppose we could, though it's only happening when a webhook is created, so the performance impact would be minor. The better case would be if it's more readable.

Reply:
@mickmister Haven't implemented this, felt it's sufficient to leave as is unless reviewers feel otherwise.